src/coreclr/CMakeLists.txt

```diff
@@ -101,7 +101,7 @@ if(CLR_CMAKE_HOST_UNIX)
       add_linker_flag(-Wl,-z,notext)
     endif()
 
-    if(NOT CLR_CMAKE_HOST_MACCATALYST AND NOT CLR_CMAKE_HOST_IOS AND NOT CLR_CMAKE_HOST_TVOS)
+    if(NOT FEATURE_NATIVEAOT_ONLY)
```

Comment:
Please revert this commit. There is a PR ready to merge to enable build of coreclr on iOS / macCatalyst that changes all of these conditions to `NOT CLR_CMAKE_HOST_TVOS` only.